pubsub/trace.go

```diff
@@ -306,71 +306,124 @@ func (c messageCarrier) Keys() []string {
 	return out
 }
 
+// injectPropagation injects context data into the Pub/Sub message's Attributes field.
+func injectPropagation(ctx context.Context, msg *Message) {
+	// only inject propagation if a valid span context was detected.
+	if trace.SpanFromContext(ctx).SpanContext().IsValid() {
+		if msg.Attributes == nil {
+			msg.Attributes = make(map[string]string)
+		}
+		propagation.TraceContext{}.Inject(ctx, newMessageCarrier(msg))
```

Comment:
Any specific reason you use `propagation.TraceContext{}` here instead of `otel.GetTextMapPropagator()`?

I have a feeling this could benefit from using the global propagator that services likely already have wired for other tracing use cases.

Reply:
For example: in our company we also wire up `propagation.Baggage{}`

Reply:
Thanks for bringing this up. It's definitely we've considered, but opted out of using the global propagator because:
1. We want to reduce the chances of user error, where someone forgets to set the propagator and context propagation breaks 
2. Currently attributes are injected into the message attributes, with the  `googclient_` prefix. These attributes have to be reserved, and passing in an undefined `googclient_foo` attribute will result in an error.

We may consider changing the propagator to a composite propagator that includes baggage in the future. This is a non-breaking change, even for users that use a combination of composite (new) and non-composite (old) propagators.